### PR TITLE
Customer module code optimisation in if/else block

### PR DESCRIPTION
**Pull request by gokulra92**
Removed unwanted else blocks and make else operation as default

### Description (*)

In some cases we don't need else case, We can move the else case as default.
    For example:
`
$status = '';
if ($age >= '18') {
 $status = 'allowed';
} else {
   $status = 'notallowed';
}
echo $status;
`
**Possible 1:**
`
$status = 'notallowed';
if ($age >= '18') {
 $status = 'allowed';
}
echo $status;
`
**Possible 2**
`
$status = ($age >= '18') ? 'allowed' : 'notallowed';
echo $status;
`
    The above both possibilities will returns the same result. It will reduce the functionality run time and number of lines.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Customer/Controller/Account/CreatePassword.php [hunk 1]
```diff
@@ -16,9 +16,9 @@
 use Magento\Framework\App\ObjectManager;
 
 /**
- * Class CreatePassword
+ * Class to create or reset the password for customer
  *
- * @package Magento\Customer\Controller\Account
+ * @author      Magento Core Team <core@magentocommerce.com>
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Controller/Account/CreatePassword.php:21]: Could you remove this line?

File: app/code/Magento/Customer/Controller/Account/CreatePost.php [hunk 2]
```diff
@@ -119,6 +119,11 @@ class CreatePost extends AbstractAccount implements CsrfAwareActionInterface, Ht
      */
     protected $session;
 
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $scopeConfig;
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Controller/Account/CreatePost.php:125]: Could you explain why you added this property?

File: app/code/Magento/Customer/Controller/Account/ForgotPasswordPost.php [hunk 3]
```diff
@@ -40,16 +46,19 @@ class ForgotPasswordPost extends \Magento\Customer\Controller\AbstractAccount im
...
      * @param Session $customerSession
      * @param AccountManagementInterface $customerAccountManagement
+     * @param LoggerInterface $logger
      * @param Escaper $escaper
      */
     public function __construct(
         Context $context,
         Session $customerSession,
         AccountManagementInterface $customerAccountManagement,
+        LoggerInterface $logger,
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Controller/Account/ForgotPasswordPost.php:56]: Could you make it last argument and make it optional with fallback to object manager? (needed for backward compatiblity)

File: app/code/Magento/Customer/Controller/Account/ForgotPasswordPost.php [hunk 4]
```diff
@@ -79,6 +88,7 @@ public function execute()
                     AccountManagement::EMAIL_RESET
                 );
             } catch (NoSuchEntityException $exception) {
+                $this->logger->info($exception);
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Controller/Account/ForgotPasswordPost.php:91]: Could you explain why do you need to log this exception and wasn't needed before?

File: app/code/Magento/Customer/Controller/Account/LoginPost.php [hunk 5]
```diff
@@ -42,6 +42,11 @@ class LoginPost extends AbstractAccount implements CsrfAwareActionInterface, Htt
      */
     protected $formKeyValidator;
 
+    /**
+     * @var CustomerUrl
+     */
+    protected $customerUrl;
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Controller/Account/LoginPost.php:48]: Could you explain why you added this property?

File: app/code/Magento/Customer/Helper/Address.php [hunk 6]
```diff
@@ -121,42 +121,6 @@ public function __construct(
...
-
-    /**
-     * Retrieve create url.
-     *
-     * @return void
-     */
-    public function getCreateUrl()
-    {
-    }
-
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Helper/Address.php:159]: Please revert this change, it's not backward compatible

File: app/code/Magento/Customer/Model/Metadata/Form/AbstractData.php [hunk 7]
```diff
@@ -98,6 +105,7 @@ public function __construct(
         \Psr\Log\LoggerInterface $logger,
         \Magento\Customer\Api\Data\AttributeMetadataInterface $attribute,
         \Magento\Framework\Locale\ResolverInterface $localeResolver,
+        \Magento\Framework\Filesystem\Driver\Http $http,
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Model/Metadata/Form/AbstractData.php:108]: Please make this parameter optional with fallback to object manger due to backward compatibility reasons

File: app/code/Magento/Customer/Model/Metadata/Form/AbstractData.php [hunk 8]
```diff
@@ -64,6 +65,11 @@ abstract class AbstractData
      */
     protected $_localeResolver;
 
+    /**
+     * @var \Magento\Framework\Filesystem\Driver\Http
+     */
+    protected $_http;
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Model/Metadata/Form/AbstractData.php:71]: Please make it private and without underscore prefix.
```suggestion
    private $http;
```

File: app/code/Magento/Customer/Model/Metadata/Form/Select.php [hunk 9]
```diff
@@ -10,18 +10,23 @@
 use Magento\Customer\Model\Metadata\ElementFactory;
 use Magento\Framework\App\RequestInterface;
 
+/**
+ * Form Element Select Data Model
+ *
+ * @author      Magento Core Team <core@magentocommerce.com>
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Model/Metadata/Form/Select.php:16]: Please remove this line

File: app/code/Magento/Customer/Ui/Component/Listing/Column/InlineEditUpdater.php [hunk 10]
```diff
@@ -7,6 +7,11 @@
 
 use Magento\Framework\View\Element\UiComponentInterface;
 
+/**
+ * Add editor config to component configuration with correct editorType
+ *
+ * @author      Magento Core Team <core@magentocommerce.com>
```
Comment (ihor-sviziev) [inline app/code/Magento/Customer/Ui/Component/Listing/Column/InlineEditUpdater.php:13]: please remove this line

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@slavvka @lenaorobei @naydav could you add info why do we have "no build reports available" on this PR?

**Comment by lenaorobei** [comment]
Here is the fatal error in log.
```
00:23:57 [Build Functional B2B] Compilation was started.
00:23:57 [Build Functional B2B] %message% 0/8 [>---------------------------]   0% < 1 sec 50.0 MiBProxies code generation... 0/8 [>------------------------]   0% < 1 sec 50.0 MiB
00:23:57 [Build Functional B2B] Proxies code generation... 1/8 [===>---------------------]  12% < 1 sec 54.0 MiB
00:23:57 [Build Functional B2B] Repositories code generation... 1/8 [==>-----------------]  12% < 1 sec 54.0 MiBPHP Fatal error:  Cannot redeclare Magento\Customer\Controller\Account\CreatePost::$scopeConfig in /var/www/html/app/code/Magento/Customer/Controller/Account/CreatePost.php on line 160 [] []
```

File: app/code/Magento/Customer/Controller/Account/CreatePost.php (hunk 2 above)
Comment (eduard13) [reply app/code/Magento/Customer/Controller/Account/CreatePost.php:125]: @gokulra92, the $scopeConfig is already declared below as `private $scopeConfig`.

Comment (ihor-sviziev) [reply app/code/Magento/Customer/Controller/Account/CreatePost.php:125]: Yeah, please remove this line

**Comment by engcom-Charlie** [comment]
Hi @gokulra92, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.
